src/backend/opencl/platform.cpp

```diff
@@ -359,8 +361,9 @@ bool isDoubleSupported(unsigned device) {
         common::lock_guard_t lock(devMngr.deviceMutex);
         dev = *devMngr.mDevices[device];
     }
-
-    return (dev.getInfo<CL_DEVICE_DOUBLE_FP_CONFIG>() > 0);
+    // 64bit fp is an optional extension
+    return (dev.getInfo<CL_DEVICE_EXTENSIONS>().find("cl_khr_fp64") !=
+            string::npos);
```

Comment:
👍🏾 

Although I wonder if the vendor implementation caches the device type support in internal cache. If cached in such manner, then fetching that compared to doing a string search is more efficient.

Reply:
I will get through the debugger step by step and report what I find.

Reply:
I have the impression that it is straight copy into a vector<char>.
For NVIDIA
- Total length of vector: 453 chars
- cl_khr_fp16 NA
- cl_khr_fp64 at pos 138
For AMD
- Total length of vector: 661 chars
- cl_khr_fp16 NA
- cl_khr_fp64 at pos 0

The complete string is cached in the driver.  Copy happens in cl2.hpp at line 1427.
Although I can not see inside the function, I come to this conclusion because the exact length is provided upfront during the construction of the vector.